cmp/form/NumberField.js

```diff
@@ -52,13 +68,44 @@ export class NumberField extends HoistField {
         if (ev.key === 'Enter') this.doCommit();
     }
 
-    toExternal(val) {
-        const ret = Number.parseFloat(val);
-        return isFinite(ret) ? ret : null;
+    toExternal(value) {
+        value = this.parseValue(value);
+        return isNaN(value) ? null : value;
     }
 
-    toInternal(val) {
-        return val != null ? val.toString() : '';
+    toInternal(value) {
+        if (value == null) return '';
+        const props = this.props,
+            precision = props.precision != null ? props.precision : 4,
```

Comment:
Trying to decide if it makes sense to have 4 as a default here, versus some larger value (16) or no limit. When accepting user entry it seems more aggressive to truncate precision entered by the user than it does in the corresponding case of formatting a number for display.

That said, we have the prop available, so a high precision use case can deliberately up it, so I am happy to leave four as a likely most-useful-default.